### PR TITLE
api: supply id for minor meaningless device

### PR DESCRIPTION
**Pull request by ferris-cx**
### Ⅰ. Describe what this PR does

The RDMA device is mounted inside the container based on the scheduling assignment result.

### Ⅱ. Does this pull request fix one issue?

The scheduling framework already supports joint allocation of Gpus and RDMA devices, but several semantics need to be tested in practice, including preference and samePCIE. In order to allow RDMA devices to access the container, we need to fix the lack of BDF addresses in the results assigned by the scheduling algorithm to RDMA devices

### Ⅲ. Describe how to verify it

1. In the k8s cluster, prepare one or more servers that support rdma network adapters as cluster nodes. Install the new version of the koordlet component, koord-manager, and the revamped multus-cni on each node, and check the node status. The number of resources is displayed as the actual number of RDMA nics on the node.

2. Write a pod.YAML to apply for RDMA network card resources, kubectl apply-f pod. On pod note (device-allocated), check whether the rdma device bdf address (busID field) is included in the rdma allocation result, and check the accuracy.

3. When the pod is in the running state, enter the container and run the ifconfig command to check the number of network cards. If the case is correct, multiple network card lists will be output, such as net1, net2, net3, and so on.

### Ⅳ. Special notes for reviews

1. Deploy components that support RDMA, including koordlet, koord-manager, koord-scheduler, and multus-cni.

2. Due to the complete end-to-end passthrough, multi-CNI plug-in is also required, which complies with CNI specifications and supports multi-NIC allocation. Assigning the RDMA network card PF/VF to the Pod mentioned here requires the device ID to be injected into the component, otherwise it will not work. This change will be maintained separately in the multi-cni project or another PR

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/scheduler/plugins/deviceshare/plugin.go [hunk 1]
```diff
@@ -545,12 +546,43 @@ func (p *Plugin) preBindObject(ctx context.Context, cycleState *framework.CycleS
 		return nil
 	}
 
+	var deviceAllocs *apiext.DeviceAllocations
```
Comment (ZiMengSheng) [inline pkg/scheduler/plugins/deviceshare/plugin.go:549]: can we consider add move this logic to Reserve() phase?

File: apis/extension/device_share.go [hunk 2]
```diff
@@ -89,6 +89,7 @@ type DeviceAllocation struct {
 	Minor     int32                      `json:"minor"`
 	Resources corev1.ResourceList        `json:"resources"`
 	Extension *DeviceAllocationExtension `json:"extension,omitempty"`
+	BusID     string                     `json:"busID,omitempty"`
```
Comment (ZiMengSheng) [inline apis/extension/device_share.go:92]: I prefer define it as this:
```
type DeviceAllocation struct {
	Minor     int32                      `json:"minor"`
	Resources corev1.ResourceList        `json:"resources"`
	Extension *DeviceAllocationExtension `json:"extension,omitempty"`
	Topology  *DeviceTopology            `json:"topology,omitempty"`
}

type DeviceTopology struct {
	// BusID is the domain:bus:device.function formatted identifier of PCI/PCIE device
	BusID string `json:"busID,omitempty"`
}
```

Comment (ZiMengSheng) [inline apis/extension/device_share.go:92]: I prefer define it as this:
```
type DeviceAllocation struct {
	Minor     int32                      `json:"minor"`
	Resources corev1.ResourceList        `json:"resources"`
	Extension *DeviceAllocationExtension `json:"extension,omitempty"`
	Topology  *DeviceTopology            `json:"topology,omitempty"`
}

type DeviceTopology struct {
	// BusID is the domain:bus:device.function formatted identifier of PCI/PCIE device
	BusID string `json:"busID,omitempty"`
}
```

Comment (ZiMengSheng) [inline apis/extension/device_share.go:92]: I prefer define it as this:
```
type DeviceAllocation struct {
	Minor     int32                      `json:"minor"`
	Resources corev1.ResourceList        `json:"resources"`
	Extension *DeviceAllocationExtension `json:"extension,omitempty"`
	Topology  *DeviceTopology            `json:"topology,omitempty"`
}

type DeviceTopology struct {
	// BusID is the domain:bus:device.function formatted identifier of PCI/PCIE device
	BusID string `json:"busID,omitempty"`
}
```

Comment (ZiMengSheng) [reply apis/extension/device_share.go:92]: and use ID store BDF

Comment (ferris-cx) [reply apis/extension/device_share.go:92]: OK！BusId is one of the attributes of the topology structure and can be encapsulated in the topology structure！

Comment (ferris-cx) [reply apis/extension/device_share.go:92]: Was the ID used for something else?

Comment (ZiMengSheng) [reply apis/extension/device_share.go:92]: Ok，cancel this

File: pkg/scheduler/plugins/deviceshare/plugin.go (hunk 1 above)
Comment (ferris-cx) [reply pkg/scheduler/plugins/deviceshare/plugin.go:549]: ok

File: apis/extension/device_share.go [hunk 3]
```diff
@@ -63,21 +63,19 @@ const (
 // DeviceAllocations would be injected into Pod as form of annotation during Pre-bind stage.
 /*
 {
-  "gpu": [
+  "rdma": [
```
Comment (songtao98) [inline apis/extension/device_share.go:66]: How about explain both gpu and rdma here and in the proposal? As a plain reader I might be confused what this annotation will look like when device type is gpu.

Comment (ZiMengSheng) [reply apis/extension/device_share.go:66]: Nice Suggesstion

**Pull request merged**